lighter/system.py

```diff
@@ -182,11 +209,25 @@ def _base_step(self, batch: Tuple, batch_idx: int, mode: str) -> Union[Dict[str,
 
                 For predict step, it returns pred only.
         """
-        input, target = batch if len(batch) == 2 else (batch[:-1], batch[-1])
+        # Split the batch into input and target.
+        if len(batch) == 1:
```

Comment:
This seems a bit unnecessary to me, can we not just have the user return the `__getitem__` as either a single element (maps to input) or two elements (maps to target)?  If not, we throw an assertion, this will make sure datasets are also consistent across projects and have a standard format that most people are familiar with.

I saw your ssl-radiologist repo and it seems like this is necessary because you expand the tensor. 
https://github.com/ibro45/ssl-radiologist-lighter/blob/45beeb09f7955825c239f36478762aa17f0dbd77/datasets/idc_ssl.py#L32
Would it not work if it was just a list of tensors instead? Collate should handle it the same way right?

Reply:
Agree - I initially allowed inputs like that because of the Lightly losses that I was using, but having it more restrictive/defined is better, can't cover all the cases. The user should override the arguments of a loss so that it's `forward(pred, target)` always.

I think I could actually write a util function that can patch a function or method to be like that by mapping the arguments from the original method/function to new arguments. What do you think? Would be a separate PR

Reply:
@surajpaib what do you think about it now

Reply:
Looks great!